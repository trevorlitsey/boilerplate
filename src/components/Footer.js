import React from 'react';
import styled from 'styled-components';
import { Link } from "react-router-dom";

const P = styled.p`
	text-align: center;
	color: grey;
`

const Footer = () => (
	<P> <a href="http://www.trevorlitsey.com" target="blank">trevorlitsey.com</a> | <a href="https://github.com/trevorlitsey/cashflow-calc" target="blank">github</a> </P>
)

export default Footer;