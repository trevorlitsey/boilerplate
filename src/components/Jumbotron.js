import React from 'react';

const Jumbotron = () => (
	<div className="jumbotron">
		<h1 style={{ textDecoration: 'underline' }} className="display-4">Welcome To Boilerplate</h1>
		<p className="lead">Drag and drop snippets to draft a new document</p>
		<hr className="my-4" />
		<p>When you're done, click the button to download text as a word document.</p>
		<a className="btn btn-primary btn-lg" href="#" role="button">Download .docx</a>
	</div>
)

export default Jumbotron;