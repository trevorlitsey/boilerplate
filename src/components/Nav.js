import React from 'react';
import FontAwesomeIcon from '@fortawesome/react-fontawesome';
import faGoogle from '@fortawesome/fontawesome-free-brands/faGoogle';
import styled from 'styled-components';

const Container = styled.div`
	width: 100%;
	display: flex;
	justify-content: space-between;
`

const Nav = () => (
	<nav className="navbar navbar-expand-lg navbar-dark bg-primary">
		<Container>
			<div className="left">
				<ul className="navbar-nav mr-auto">
					<a className="navbar-brand" href="#">~Boilerplate Rocks~</a>
					<li className="nav-item active">
						<a className="nav-link" href="#">Preview<span className="sr-only">(current)</span></a>
					</li>
					<li className="nav-item">
						<a className="nav-link" href="#">Snippets</a>
					</li>
				</ul>
			</div>
			<div className="right">
				<button type="button" className="btn btn-light"><FontAwesomeIcon style={{ marginRight: '6px' }} icon={faGoogle} />Sign in with Google</button>
			</div>
		</Container>
	</nav >
)



export default Nav;