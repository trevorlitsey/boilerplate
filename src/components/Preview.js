import React from 'react';
import styled from 'styled-components';

import { convertObjToArr } from './helpers';

const Container = styled.div`

	display: grid;
	grid-template-columns: 1fr 3fr;

	.scrollspy {
		padding: 0 20px;
		position: relative;
		height: 500px;
		overflow-y: scroll;
	}
`

const Link = ({ href, text, active }) => (
	<a className={`list-group-item list-group-item-action ${active && 'active'}`} href={href} > {text}</a>
)

class Preview extends React.Component {

	state = {
		links: [
			{
				order: 1,
				href: '#list-item-1',
				active: true,
			},
			{
				order: 2,
				href: '#list-item-2',
				active: false,
			},
			{
				order: 3,
				href: '#list-item-3',
				active: false,
			},
			{
				order: 4,
				href: '#list-item-4',
				active: false,
			},
		],
		recentClick: false,
	}

	handleScroll = (e) => {

		const links = [...this.state.links];

		const containerScroll = this.refs['container'].scrollTop;
		const containerHeight = this.refs['container'].clientHeight;
		const containerScrollHeight = this.refs['container'].scrollHeight;

		let paraFound = false;

		if (containerScroll + containerHeight >= containerScrollHeight - 5) { // we have reached the bottom
			links.forEach(link => link.active = false);
			links[links.length - 1].active = true;
		} else {
			links
				.reverse() // search from bottom up
				.forEach((link) => {

					const { href } = link
					const elOffsetTop = this.refs[href].offsetTop;
					const elOffsetHeight = this.refs[href].offsetHeight;

					if (elOffsetTop < containerScroll + 5 && !paraFound) {
						link.active = true;
						paraFound = true;
					} else {
						link.active = false;
					}
				})
			links.reverse()
		}

		this.setState({ links })
	}

	render() {

		const { links } = this.state;

		return (
			<Container>
				<div id="list-example" className="list-group">
					{links.map(({ href, active }) => <Link key={href} href={href} text={href} active={active} />)}
				</div>
				<div ref="container" onScroll={this.handleScroll} data-spy="scroll" data-target="#list-example" data-offset="0" className="scrollspy scrollspy-example">
					<h4 ref="#list-item-1" id="list-item-1">Item 1</h4>
					<p>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aenean venenatis et ipsum id dapibus. Vestibulum sit amet posuere odio, sit amet tincidunt mauris. Donec turpis massa, venenatis sed maximus quis, luctus a quam. In hac habitasse platea dictumst. Praesent nec justo egestas, feugiat nunc in, dapibus enim. Fusce finibus eros quis metus vulputate imperdiet. Morbi sed urna ut elit dignissim pharetra. Vivamus vitae massa quis nisi dignissim consequat ac vitae sapien.Donec placerat neque vitae blandit pretium. Donec risus metus, feugiat ac ultrices quis, sagittis eget mauris. Aenean lobortis posuere tellus vitae tincidunt. Sed cursus congue ipsum, nec laoreet elit. Sed a euismod erat, id fringilla odio. Donec ultricies commodo nibh, eu fringilla libero hendrerit id. Quisque fermentum ipsum sit amet risus tempus vulputate.</p>
					<p>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aenean venenatis et ipsum id dapibus. Vestibulum sit amet posuere odio, sit amet tincidunt mauris. Donec turpis massa, venenatis sed maximus quis, luctus a quam. In hac habitasse platea dictumst. Praesent nec justo egestas, feugiat nunc in, dapibus enim. Fusce finibus eros quis metus vulputate imperdiet. Morbi sed urna ut elit dignissim pharetra. Vivamus vitae massa quis nisi dignissim consequat ac vitae sapien.Donec placerat neque vitae blandit pretium. Donec risus metus, feugiat ac ultrices quis, sagittis eget mauris. Aenean lobortis posuere tellus vitae tincidunt. Sed cursus congue ipsum, nec laoreet elit. Sed a euismod erat, id fringilla odio. Donec ultricies commodo nibh, eu fringilla libero hendrerit id. Quisque fermentum ipsum sit amet risus tempus vulputate.</p>
					<h4 ref="#list-item-2" id="list-item-2">Item 2</h4>
					<p>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aenean venenatis et ipsum id dapibus. Vestibulum sit amet posuere odio, sit amet tincidunt mauris. Donec turpis massa, venenatis sed maximus quis, luctus a quam. In hac habitasse platea dictumst. Praesent nec justo egestas, feugiat nunc in, dapibus enim. Fusce finibus eros quis metus vulputate imperdiet. Morbi sed urna ut elit dignissim pharetra. Vivamus vitae massa quis nisi dignissim consequat ac vitae sapien....</p>
					<h4 ref="#list-item-3" id="list-item-3">Item 3</h4>
					<p>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aenean venenatis et ipsum id dapibus. Vestibulum sit amet posuere odio, sit amet tincidunt mauris. Donec turpis massa, venenatis sed maximus quis, luctus a quam. In hac habitasse platea dictumst. Praesent nec justo egestas, feugiat nunc in, dapibus enim. Fusce finibus eros quis metus vulputate imperdiet. Morbi sed urna ut elit dignissim pharetra. Vivamus vitae massa quis nisi dignissim consequat ac vitae sapien.Donec placerat neque vitae blandit pretium. Donec risus metus, feugiat ac ultrices quis, sagittis eget mauris. Aenean lobortis posuere tellus vitae tincidunt. Sed cursus congue ipsum, nec laoreet elit. Sed a euismod erat, id fringilla odio. Donec ultricies commodo nibh, eu fringilla libero hendrerit id. Quisque fermentum ipsum sit amet risus tempus vulputate....</p>
					<h4 ref="#list-item-4" id="list-item-4">Item 4</h4>
					<p>Donec placerat neque vitae blandit pretium. Donec risus metus, feugiat ac ultrices quis, sagittis eget mauris. Aenean lobortis posuere tellus vitae tincidunt. Sed cursus congue ipsum, nec laoreet elit. Sed a euismod erat, id fringilla odio. Donec ultricies commodo nibh, eu fringilla libero hendrerit id. Quisque fermentum ipsum sit amet risus tempus vulputate....</p>
					<p>Donec placerat neque vitae blandit pretium. Donec risus metus, feugiat ac ultrices quis, sagittis eget mauris. Aenean lobortis posuere tellus vitae tincidunt. Sed cursus congue ipsum, nec laoreet elit. Sed a euismod erat, id fringilla odio. Donec ultricies commodo nibh, eu fringilla libero hendrerit id. Quisque fermentum ipsum sit amet risus tempus vulputate....</p>
				</div>
			</Container>
		)
	}
}

export default Preview;