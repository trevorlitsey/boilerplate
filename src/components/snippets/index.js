export { default as SearchBar } from './SearchBar';
export { default as SnippetWell } from './SnippetWell';
export { default as NewSnippetButton } from './NewSnippetButton';
export { default as SnippetModal } from './SnippetModal';
export { default as SignInMessage } from './SignInMessage'