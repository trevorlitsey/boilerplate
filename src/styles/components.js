import styled from 'styled-components';

export const XScroll = styled.div`
  overflow-x: scroll;
`;

export const H4 = styled.h4`
  text-decoration: underline;
  margin-bottom: 0.75rem;
`;

export const GrayH4 = styled.h4`
  color: var(--gray);
  textalign: center;
  margin: 8px;
  text-align: center;
  width: 100%;
`;

export const H3 = styled.h3`
  text-decoration: underline;
  margin-bottom: 0.75rem;
`;
