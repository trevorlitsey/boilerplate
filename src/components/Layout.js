import React from 'react';
import { object } from 'prop-types';
import styled from 'styled-components';

import Nav from './Nav';

const Wrapper = styled.div`
	max-width: 1600px;
	margin: auto;
	padding: 40px;
`

const Layout = ({ location, children }) => (
	<div>
		<Nav location={location} />
		<Wrapper>
			{children}
		</Wrapper>
	</div>
)

Layout.propTypes = {
	location: object.isRequired,
}

export default Layout