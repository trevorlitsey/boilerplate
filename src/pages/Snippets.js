import React from 'react';
import styled from 'styled-components';
import uniqid from 'uniqid';

import firebase, { db } from '../firebase';

import { Layout } from '../components/layout';
import { Spinner } from '../components/shared';
import { SearchBar, SnippetWell, NewSnippetButton, SnippetModal, SignInMessage } from '../components/snippets';

import { H4 } from '../styles/components';

class Snippets extends React.PureComponent {

	state = {
		user: this.props.user || {},
		modalOn: false,
		snippets: {},
		tags: [],
		snippetToEdit: '',
		filterQuery: '',
		userLoaded: this.props.userLoaded || false,
		dbLoaded: false,
	}

	componentDidMount = () => {

		firebase.auth().onAuthStateChanged((user) => {
			this.setState({ user, userLoaded: true });

			if (user) {
				this.dbRef = db.doc(`users/${user.uid}/`)
				this.dbUnsubscribe = this.dbRef
					.onSnapshot((doc) => {
						if (doc.data()) {
							const { snippets, tags } = doc.data();
							this.setState({ snippets, tags });
						}
						this.setState({ dbLoaded: true });
					});
			} else {
				this.dbUnsubscribe && this.dbUnsubscribe();
			}
		})
	}

	componentWillUnmount = () => {
		this.dbUnsubscribe && this.dbUnsubscribe();
		this.dbRef = null;
	}

	showModal = (e, snippetToEdit = '') => {
		this.setState({ modalOn: true, snippetToEdit })
	}

	hideModal = () => {
		this.setState({ modalOn: false, snippetToEdit: '' })
	}

	addSnippet = (newSnippet, id = uniqid()) => {
		const snippets = { ...this.state.snippets }
		snippets[id] = newSnippet;
		this.dbRef.set({ snippets }, { merge: true });
	}

	updateSnippet = (updatedSnippet) => {
		const { snippets, snippetToEdit } = { ...this.state }
		snippets[snippetToEdit] = updatedSnippet;
		this.dbRef.set({ snippets }, { merge: true });
	}

	deleteSnippet = (id) => {
		const snippets = { ...this.state.snippets }
		delete snippets[id];
		this.hideModal();
		this.dbRef.update({ snippets: firebase.firestore.FieldValue.delete() }); // make sure that fucker is gone...

		if (Object.keys(snippets)) {
			this.dbRef.set({ snippets }, { merge: true });
		}
	}

	addTag = (newTag, id = uniqid()) => {
		const tags = [...this.state.tags];
		newTag.id = id;
		tags.push(newTag);
		this.dbRef.set({ tags }, { merge: true });
	}

	handleSearchChange = (e) => {
		this.setState({ filterQuery: e.target.value.toLowerCase() });
	}

	render() {

		const { location } = this.props;
		const {
			modalOn,
			snippets,
			tags,
			snippetToEdit,
			user,
			filterQuery,
			userLoaded,
			dbLoaded } = this.state;

		if (!userLoaded) {
			return (
				<Layout location={location} data-test="loading">
					<Spinner />
				</Layout>
			)
		}

		if (userLoaded && (!user || !Object.keys(user).length)) {
			return (
				<Layout location={location} user={user}>
					<SignInMessage />
				</Layout>
			)
		}

		return (
			<Layout location={location} user={user} data-test="render-page">
				<SearchBar onChange={this.handleSearchChange} />
				<SnippetWell
					snippets={snippets}
					filterQuery={filterQuery}
					showModal={this.showModal}
					loading={!dbLoaded}
				/>
				<NewSnippetButton showModal={this.showModal} />
				<SnippetModal
					snippetToEdit={snippetToEdit && snippets ? { ...snippets[snippetToEdit], id: snippetToEdit } : {}}
					modalOn={modalOn}
					hideModal={this.hideModal}
					tags={tags}
					addTag={this.addTag}
					addSnippet={this.addSnippet}
					updateSnippet={this.updateSnippet}
					deleteSnippet={this.deleteSnippet}
				/>
			</Layout>
		)
	}
}

export default Snippets;