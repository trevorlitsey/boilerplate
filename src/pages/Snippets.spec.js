import React from 'react';
import { shallow } from 'enzyme';
import uniqId from 'uniqid';

import { tags } from '../../sampleData';

import { SignInMessage, SnippetWell } from '../components/snippets';
import Snippets from './Snippets';

describe('Snippets', () => {

	it('should render show spinner if user is loading', () => {
		const wrapper = renderSnippets({ userLoaded: false });
		expect(wrapper.find('[data-test="loading"]')).toHaveLength(1);
		expect(wrapper).toMatchSnapshot();
	})

	it('should display signIn message if user is not logged in', () => {
		const props = { userLoaded: true, user: null }
		const wrapper = renderSnippets(props);
		expect(wrapper.find(SignInMessage)).toHaveLength(1);
		expect(wrapper).toMatchSnapshot();
	})

	it('should render page if user is signed in', () => {
		const props = {
			userLoaded: true,
			user: { uid: '57uhs2whsjj' },
		}
		const wrapper = renderSnippets(props);
		expect(wrapper.find(SnippetWell)).toHaveLength(1);
		expect(wrapper).toMatchSnapshot();
	})

	it('should showModal', () => {
		const instance = renderSnippets().instance();
		expect(instance.state.modalOn).toBe(false);
		instance.showModal()
		expect(instance.state.modalOn).toBe(true);
	})

	it('should hideModal', () => {
		const instance = renderSnippets().instance();
		instance.setState({ modalOn: true })
		expect(instance.state.modalOn).toBe(true);
		instance.hideModal()
		expect(instance.state.modalOn).toBe(false);
	})

	it('should addSnippet', () => {
		const instance = renderSnippets().instance();
		instance.dbRef = { set: jest.fn() }
		const newSnippet = { title: 'this is a title', text: 'this is some text' }
		const id = uniqId();
		instance.addSnippet(newSnippet, id);

		const expectedSnippets = {
			[id]: newSnippet,
		}
		expect(instance.dbRef.set).toHaveBeenCalledWith({ snippets: expectedSnippets }, { merge: true })
	})

	it('should updateSnippet', () => {
		// setup
		const instance = renderSnippets().instance();
		instance.dbRef = { set: jest.fn() };
		const snippetToEdit = '276tghns'
		const snippets = { [snippetToEdit]: { title: 'what a title', text: 'what a text' } };
		instance.setState({ snippets, snippetToEdit });

		// confirm setup
		expect(instance.state.snippets).toEqual(snippets);

		// update
		const newTitle = 'this is the new title';
		const updatedSnippet = { title: newTitle, text: 'what a text' }
		instance.updateSnippet(updatedSnippet);

		// expect
		const expectedUpdatedSnippets = { [snippetToEdit]: updatedSnippet };
		expect(instance.dbRef.set).toHaveBeenCalledWith({ snippets: expectedUpdatedSnippets }, { merge: true })
	})

	it('should deleteSnippet', () => {
		// setup
		const instance = renderSnippets().instance();
		instance.dbRef = { update: jest.fn(), set: jest.fn() }
		const idToDelete = '87uhsjs';
		const snippets = { [idToDelete]: { title: 'what a title', text: 'what a text' } }
		instance.setState({ snippets });

		// confirm setup
		expect(instance.state.snippets).toEqual(snippets)

		// delete
		instance.deleteSnippet(idToDelete);

		// expect
		expect(instance.dbRef.update).toHaveBeenCalled();
		expect(instance.dbRef.set).toHaveBeenCalledWith({ snippets: {} }, { merge: true });
	})

	it('should addTag', () => {
		// setup
		const instance = renderSnippets().instance();
		instance.dbRef = { set: jest.fn() }
		instance.setState({ tags });
		const id = uniqId();
		const newTag = { id, label: 'new tag!', value: 'new tag!' }

		// add tag
		instance.addTag(newTag, id);

		// expect
		instance.state.tags.push(newTag); // TODO, find a better way to do this
		expect(instance.state.tags).toHaveLength(4);
		expect(instance.dbRef.set).toHaveBeenCalledWith({ tags: instance.state.tags }, { merge: true })
	})

	it('should handleSearchChange', () => {
		// setup
		const instance = renderSnippets().instance();
		const newFilterQuery = 'this is a search query';
		const event = { target: { value: newFilterQuery } }

		// update
		instance.handleSearchChange(event);

		//expect
		expect(instance.state.filterQuery).toBe(newFilterQuery)
	})

})

// ---------
function renderSnippets(props = {}) {
	const propsToUser = {
		user: { uid: '2767ushs' },
		userLoaded: true,
		...props,
	}

	return shallow(<Snippets {...propsToUser} />)
}