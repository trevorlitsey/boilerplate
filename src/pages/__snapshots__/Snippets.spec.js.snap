// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snippets should display signIn message if user is not logged in 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Snippets
    user={null}
    userLoaded={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <SignInMessage />,
      "location": undefined,
      "user": Object {},
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <SignInMessage />,
        "location": undefined,
        "user": Object {},
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Snippets should render page if user is signed in 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Snippets
    user={
      Object {
        "uid": "57uhs2whsjj",
      }
    }
    userLoaded={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <SearchBar
          onChange={[Function]}
        />,
        <SnippetWell
          filterQuery=""
          loading={true}
          showModal={[Function]}
          snippets={Object {}}
        />,
        <NewSnippetButton
          showModal={[Function]}
        />,
        <SnippetModal
          addSnippet={[Function]}
          addTag={[Function]}
          deleteSnippet={[Function]}
          hideModal={[Function]}
          modalOn={false}
          snippetToEdit={Object {}}
          tags={Array []}
          updateSnippet={[Function]}
        />,
      ],
      "data-test": "render-page",
      "location": undefined,
      "user": Object {
        "uid": "57uhs2whsjj",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "onChange": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "filterQuery": "",
          "loading": true,
          "showModal": [Function],
          "snippets": Object {},
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "showModal": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "addSnippet": [Function],
          "addTag": [Function],
          "deleteSnippet": [Function],
          "hideModal": [Function],
          "modalOn": false,
          "snippetToEdit": Object {},
          "tags": Array [],
          "updateSnippet": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <SearchBar
            onChange={[Function]}
          />,
          <SnippetWell
            filterQuery=""
            loading={true}
            showModal={[Function]}
            snippets={Object {}}
          />,
          <NewSnippetButton
            showModal={[Function]}
          />,
          <SnippetModal
            addSnippet={[Function]}
            addTag={[Function]}
            deleteSnippet={[Function]}
            hideModal={[Function]}
            modalOn={false}
            snippetToEdit={Object {}}
            tags={Array []}
            updateSnippet={[Function]}
          />,
        ],
        "data-test": "render-page",
        "location": undefined,
        "user": Object {
          "uid": "57uhs2whsjj",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "onChange": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "filterQuery": "",
            "loading": true,
            "showModal": [Function],
            "snippets": Object {},
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "showModal": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "addSnippet": [Function],
            "addTag": [Function],
            "deleteSnippet": [Function],
            "hideModal": [Function],
            "modalOn": false,
            "snippetToEdit": Object {},
            "tags": Array [],
            "updateSnippet": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Snippets should render show spinner if user is loading 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Snippets
    user={
      Object {
        "uid": "2767ushs",
      }
    }
    userLoaded={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <Spinner />,
      "data-test": "loading",
      "location": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <Spinner />,
        "data-test": "loading",
        "location": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
